tests/testthat/test-nest.R

```diff
@@ -90,26 +90,157 @@ test_that("can nest multiple columns", {
 })
 
 test_that("nesting no columns nests all inputs", {
-  # included only for backward compatibility
   df <- tibble(a1 = 1, a2 = 2, b1 = 1, b2 = 2)
-  expect_snapshot(out <- nest(df))
+  out <- nest(df)
   expect_named(out, "data")
   expect_equal(out$data[[1]], df)
 })
 
+test_that("can control output column name when nesting all inputs", {
+  df <- tibble(a1 = 1, a2 = 2, b1 = 1, b2 = 2)
+  out <- nest(df, .key = "foo")
+  expect_named(out, "foo")
+  expect_equal(out$foo[[1]], df)
+})
+
+test_that("can control output column name when only supplying `.by`", {
+  df <- tibble(a1 = 1, a2 = 2, b1 = 1, b2 = 2)
+  out <- nest(df, .by = a2, .key = "foo")
+  expect_named(out, c("a2", "foo"))
+  expect_equal(out$foo[[1]], df[c("a1", "b1", "b2")])
+})
+
+test_that("can control output column name when nesting by groups", {
+  df <- dplyr::group_by(tibble(x = c(1, 1, 1), y = 1:3), x)
+  out <- nest(df, .key = "y")
+  expect_named(out, c("x", "y"))
+})
+
+test_that("can nest `.by` columns", {
+  df <- tibble(x = c(1, 1, 1, 2, 2), y = c(2, 1, 2, 3, 4), z = 1:5)
+
+  expect_identical(
+    nest(df, .by = c(x, y)),
+    nest(df, data = z)
+  )
+})
+
+test_that("can combine `.by` with `...`", {
+  df <- tibble(x = c(1, 1, 1, 2, 2), y = c(2, 1, 2, 3, 4), z = 1:5)
+
+  expect_identical(
+    nest(df, data = x, .by = y),
+    nest(dplyr::select(df, -z), data = x)
+  )
+})
+
 test_that("nest disallows renaming", {
   df <- tibble(x = 1)
 
   expect_snapshot(error = TRUE, {
     nest(df, data = c(a = x))
   })
+  expect_snapshot(error = TRUE, {
+    nest(df, .by = c(a = x))
+  })
+})
+
+test_that("catches when `...` overwrites an existing column", {
+  df <- tibble(x = 1, y = 2)
+
+  # Hardcoded as an error.
+  # Name repair would likely break internal usage of `chop()`.
+  expect_snapshot(error = TRUE, {
+    nest(df, x = y)
+  })
 })
```

Comment:
This was previously not being caught, and was giving an even more obscure error from `chop()`